tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +171,454 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_compute(ctx, subarray_partitioner);
+ * @endcode
+ *
+ * @param The partitioner for which to compute the complete series of (sub)
+ * partitions
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_compute(
+    tiledb_ctx_t* ctx, tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Gets number of computed partitions available to be retrieved via
+ * tiledb_subarray_partitioner_get_partition().
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_compute(ctx, subarray_partitioner);
+ * @endcode
+ *
+ * @param layout The layout to set into the subarray partioner object's
+ * subarray.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_get_partition_num(
+    tiledb_ctx_t* ctx,
+    uint64_t* num,
+    tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Retrieve the part-id(th) partition subarray from the partitioner's computed
+ * subarray partitions.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint64_t part_id; //# of partition to retrieve, must be less than # computed
+ * partitions tiledb_subarray_t *retrieved_subarray;
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_get_partition(ctx, subarray_partitioner,
+ * partition_id, retrieved_subarray);
+ * @endcode
+ *
+ * @param part_id The index of a partition to retrieve from a prior
+ * tiledb_subarray_partitioner_compute(), must be less than the number
+ * of computed partitions.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ *
+ * @note Any returned subarray object was allocated within this call and is to
+ * be freed by the caller.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_get_partition(
+    tiledb_ctx_t* ctx,
+    tiledb_subarray_partitioner_t* partitioner,
+    uint64_t part_id,
+    tiledb_subarray_t* subarray);
+
+/**
+ * Set memory partitioning budget (in bytes) for fixed size attribute/dimension.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_t *partitioner;
+ * tiledb_ctx_t *ctx;
+ * uint64_t budget;
+ * char* attrname;
+ * tiledb_subarray_partitioner_set_result_budget(ctx, attrname, budget,
+ * partitioner);
+ * @endcode
+ *
+ * @param budget - byte count specifying memory budget for the specified
+ * attribute.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_result_budget(
+    tiledb_ctx_t* ctx,
+    const char* attrname,
+    uint64_t budget,
+    tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Gets result size budget (in bytes) for the input fixed-sized
+ * attribute/dimension.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_t *partitioner;
+ * tiledb_ctx_t *ctx;
+ * uint64_t *budget;
```

Comment:
uint64_t budget; and pass &budget to the function. Adjust all other places in this file for similar cases.